packages/ButtonGroup/src/components/ButtonItem/ButtonItem.js

```diff
@@ -0,0 +1,81 @@
+import React from "react";
+import PropTypes from "prop-types";
+import ButtonGroupContext from "../../ButtonGroupContext";
+
+import * as sc from "./ButtonItem.styles";
+
+const propTypes = {
+  /** Content label of the button to be displayed. */
+  children: PropTypes.node,
+
+  /** If the item is active or on selected state */
+  defaultIsActive: PropTypes.bool,
+
+  /** tabindex attribute of this item */
+  tabIndex: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+
+  /** Unique key to represent the selected value. */
+  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
+};
+
+const defaultProps = {
+  children: null,
+  defaultIsActive: false,
+  tabIndex: null,
+};
+
+const ButtonItem = props => {
+  const { children, defaultIsActive, tabIndex, value, ...moreProps } = props;
+  const {
+    currentFocusValue,
+    isDisabled,
+    isMulti,
+    hasIcon,
+    isSemantic,
+    onClick,
+    size,
+    selectedItems,
+    setSelectedItems,
+  } = React.useContext(ButtonGroupContext);
+
+  React.useEffect(() => {
+    if (defaultIsActive) {
+      setSelectedItems(prevItems => {
+        return isMulti ? [...prevItems, value] : [value];
+      });
+    }
+  }, []);
```

Comment:
An appropriate way to push the `defaultIsActive` prop up to the context for initial selected state?

Reply:
If going this way, better to use a layout effect to remove the possibility of the user seeing the wrong UI.
Have you considered having a `defaultValue` on the button group instead?

Reply:
I think having `defaultIsActive` on the actual `<Item>` is a more intuitive + desirable API, which is why I wanted to go this direction.

Reply:
My thinking is that if it's likely that the set of values is not hardcoded but rather comes from a data source, then it's more work to distribute those values across the individual items, rather than just pass it over to the root component.